ha_client.py

```diff
@@ -43,12 +43,15 @@
                       r':(?:[0-9a-fA-F]{1,4}:){,6}(?:[0-9a-fA-F]{1,4})))')
 
 
-def check_url(ip_address):
+def check_url(ip_address: str) -> str:
```

Comment:
You might already be aware of this - type checking in Python is not enforced. 

So if it really matters, you might want to extend the conditional below to check:
```python
isinstance(ip_address, str):
```

Reply:
Yep, I am aware of python static variable linking and I use it in my others projects (exactly this format - works with mypy). For HA skill, I will add it in one of next PR to all functions, so its easier to see what they do.

Instead of `isinstance` I use `if not ip_address:` This check if its not `None` (should not occure) and also if string is not empty.